### PR TITLE
Add support for AWS_IAM in EnvironmentVaultConfiguration

### PR DESCRIPTION
**Pull request by sfjverbarg**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@sfjverbarg Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-vault&pullRequestId=422)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-vault&pullRequestId=422) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: spring-vault-core/src/main/java/org/springframework/vault/config/EnvironmentVaultConfiguration.java [hunk 1]
```diff
@@ -326,6 +336,67 @@ protected ClientAuthentication awsEc2Authentication() {
...
+
+		private static AWSCredentialsProvider getAwsCredentialsProvider() {
+
+			DefaultAWSCredentialsProviderChain backingCredentialsProvider = DefaultAWSCredentialsProviderChain
+					.getInstance();
+
+			// Eagerly fetch credentials preventing lag during the first, actual login.
+			AWSCredentials firstAccess = backingCredentialsProvider.getCredentials();
+
+			AtomicReference<AWSCredentials> once = new AtomicReference<>(firstAccess);
```
Comment (mp911de) [inline spring-vault-core/src/main/java/org/springframework/vault/config/EnvironmentVaultConfiguration.java]: Care to elaborate the need for intermediate caching? Typically, `AWSCredentialsProvider` perform caching on their own unless calling `.refresh()`. We should either leave these defaults or use an entirely-static `AWSCredentials` instance.

**Comment by mp911de** [comment]
Any update on this pull request?

**Comment by mp911de** [comment]
Closing the PR without merge as we never received any feedback.

**Pull request closed without merging**